rust/garbage_collector/tests/garbage_collector_prop_test.rs

```diff
@@ -284,4 +353,156 @@ mod tests {
             })?;
         }
     }
+
+    proptest! {
+        #[test]
+        fn test_k8s_integration_gc_version_properties(
+            num_records in 50..100usize,
+            num_batches in 2..5usize,
+            cutoff_hours in 0..24i64,
```

Comment:
This prop test is not really ideal. It randomly generates data and does a timing based approach. We should be use a state machine proptest - https://proptest-rs.github.io/proptest/proptest/state-machine.html. This will allow deterministic interleaving of the various events in the system.